### PR TITLE
Add linguist-generated to gitattributes

### PR DESCRIPTION
**Pull request by pgherveou**
This will keep generated weights.rs from displaying in diffs by default
see https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github
for more details

### CODE REVIEW

**Comment by ggwpez** [comment]
Then how do we review them?

**Comment by ggwpez** [comment]
>That's the theory, can try to test this out with this PR to make sure that works as intended

Less trust, more truth. Let's go!

**Comment by ggwpez** [comment]
bot bench $ pallet dev pallet_balances

**Comment by pgherveou** [comment]
> Member

noobs question, would that bot call actually create a diff, since this PR has no changes?
I can also just touch a file by hand

**Comment by ggwpez** [comment]
> noobs question, would that bot call actually create a diff, since this PR has no changes? 

Yes it should. And yes; modifying by hand is also possible, but I am lazy and then you would have to revert it :stuck_out_tongue:   
This way we can see that it works end-to-end.

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1,4 @@
 Cargo.lock linguist-generated=true
 /.gitlab-ci.yml filter=ci-prettier
 /scripts/ci/gitlab/pipeline/*.yml filter=ci-prettier
+frame/*/src/weights.rs linguist-generated=true
```
Comment (ggwpez) [inline .gitattributes:4]:
```suggestion
frame/**/src/weights.rs linguist-generated=true
```
There are two testing/examples pallets as well.

**Review by ggwpez — APPROVED** [review]
This is how it looks now, seems fine :+1: 
image

**Review by bkchr — APPROVED** [review]

**Comment by athei** [comment]
> and they don't affect the PR diff numbers, (so people are not scared by a big number when they review your PR)

This part seems to be wrong :( Look at the diff stats of this PR. It includes the generated file. Other PRs which include this change  also have scary diff stats: #13312

The description of the feature also never mentions that it will be ignored in PR diff stats. Only in the PRs repository's language statistics. So it does seem to be intentional. I wish there were a way to hide it from PR diff stats, though.

**Comment by pgherveou** [comment]
> > and they don't affect the PR diff numbers, (so people are not scared by a big number when they review your PR)
> 
> This part seems to be wrong :( Look at the diff stats of this PR. It includes the generated file. Other PRs which include this change also have scary diff stats: #13312
> 
> The description of the feature also never mentions that it will be ignored in PR diff stats. Only in the PRs repository's language statistics. So it does seem to be intentional. I wish there were a way to hide it from PR diff stats, though.

yep saw that as well, this is not that useful after all...

**Pull request merged**